go/libraries/doltcore/branch_control/branch_control.go

```diff
@@ -228,28 +256,58 @@ func CanCreateBranch(ctx context.Context, branchName string) error {
 // However, not all CLI commands use *sql.Context, and therefore will not have any user associated with the context. In
 // these cases, CanDeleteBranch will pass as we want to allow all local commands to freely delete branches.
 func CanDeleteBranch(ctx context.Context, branchName string) error {
-	if !enabled {
-		return nil
-	}
 	branchAwareSession := GetBranchAwareSession(ctx)
 	// A nil session means we're not in the SQL context, so we allow the delete operation
 	if branchAwareSession == nil {
 		return nil
 	}
-	StaticController.Access.RWMutex.RLock()
-	defer StaticController.Access.RWMutex.RUnlock()
+	controller := branchAwareSession.GetController()
+	// Any context that has a non-nil session should always have a non-nil controller, so this is an error
+	if controller == nil {
+		return ErrMissingController.New()
+	}
+	controller.Access.RWMutex.RLock()
+	defer controller.Access.RWMutex.RUnlock()
 
 	user := branchAwareSession.GetUser()
 	host := branchAwareSession.GetHost()
 	// Get the permissions for the branch, user, and host combination
-	_, perms := StaticController.Access.Match(branchName, user, host)
+	_, perms := controller.Access.Match(branchName, user, host)
 	// If the user has the write or admin flags, then we allow access
 	if (perms&Permissions_Write == Permissions_Write) || (perms&Permissions_Admin == Permissions_Admin) {
 		return nil
 	}
 	return ErrCannotDeleteBranch.New(user, host, branchName)
 }
 
+// AddAccessEntryForContext adds an entry in the access table for the user represented by the given context. If the
+// context is missing some functionality that is needed to perform the addition, such as a user or the Controller, then
+// this simply returns.
+func AddAccessEntryForContext(ctx context.Context, branchName string) error {
```

Comment:
This adds Admin permissions if not present, so the name should reflect that

Maybe `GrantBranchAdmin`